python/lsst/dbb/buffmngrs/endpoint/ingester/plugins.py

```diff
@@ -72,46 +72,40 @@ class Gen2Ingest(Plugin):
     See ``lsst.pipe.tasks.ingest`` for details.
     """
 
+    defaults = {
+        "create": False,
+        "dryrun": False,
+        "ignoreIngested": False,
+        "mode": "link",
+        "task": "lsst.pipe.tasks.ingest.IngestTask"
+    }
+
     def __init__(self, config):
-        # Set default values for configuration settings, but override
-        # them with the ones provided at runtime, if necessary.
-        self.config = {
-            "dryrun": False,
-            "mode": "link",
-            "create": False,
-            "ignoreIngested": False,
-        }
-        self.config.update(config)
+        try:
+            butler_config = config["butler"]
+        except KeyError:
+            msg = "invalid configuration: Butler settings are missing"
+            logger.error(msg)
+            raise ValueError(msg)
```

Comment:
I'll add `from` part. I'm switching the exception type to put focus on the fact that it's about invalid configuration (wrong value if you will) and not that the configuration is stored/used internally as a dictionary. If you think I should stick to the original type or use one different than `ValueError`, I'm open for suggestions.

Reply:
Based upon concepts instead of thinking about how they are implemented, I try to use KeyError when looking something up and it is missing and ValueError when given the value and it is wrong.  In the error message, I focused on the "are missing" hence asking why switched from KeyError, whereas the "invalid configuration" could be a very generic ValueError.  That's the reason behind my comment.  Based upon your comment, it almost sounds like the code should have its own ConfigError (or whatever you would name it) class. 
 I don't have a strong enough opinion in this case to strongly push a particular solution.

Reply:
I issue a KeyError because Config is dict-like and failing to find a key in a Config is clearly a KeyError.

Reply:
All right, I'll stick to `KeyError` then. :slightly_smiling_face: